### PR TITLE
MOB-461 : TP/SL cleanup & testing

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Linear Ticket: [MOB-461 : TP/SL cleanup & testing](https://linear.app/dydx/issue/MOB-461/tpsl-cleanup-and-testing)

<br/>

## Description / Intuition
- fixes issue where limit price was getting set even when it was not originally set
- hides SL/TP UI behind feature flag
- hides SL/TP custom limit price behind separate feature flag
- pre-populates custom amount when both sl/tp orders have same custom amount, or if there is a single sl/tp and it has a custom amount

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/e2628b6e-07ae-40ae-9847-ccf4fa47201f"> |

<br/>

### Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift [hunk 1]
```diff
@@ -78,69 +78,78 @@ class dydxMarketPositionViewPresenter: HostedViewPresenter<dydxMarketPositionVie
...
-                viewModel?.stopLossStatusViewModel = .init(
-                    triggerSide: .stopLoss,
-                    triggerPriceText: dydxFormatter.shared.raw(number: stopLossOrder.triggerPrice?.doubleValue, digits: decimalDigits),
-                    limitPrice: stopLossOrder.type == .stoplimit ? dydxFormatter.shared.raw(number: stopLossOrder.price, digits: decimalDigits) : nil,
-                    // don't show amount unless order size is custom
-                    amount: position.size?.current?.doubleValue.magnitude == stopLossOrder.size.magnitude ? nil : dydxFormatter.shared.raw(number: stopLossOrder.size, digits: stepSizeDecimals),
-                    action: routeToTakeProfitStopLossAction)
+            let takeProfitOrders = triggerOrders.filter { (order: SubaccountOrder) in
+                order.marketId == position.id
+                && (order.type == .takeprofitmarket || (order.type == .takeprofitlimit && AbacusStateManager.shared.environment?.featureFlags.isSlTpLimitOrdersEnabled == true))
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift:97]: added this ff check

File: dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift [hunk 2]
```diff
@@ -78,69 +78,78 @@ class dydxMarketPositionViewPresenter: HostedViewPresenter<dydxMarketPositionVie
...
+                        triggerPriceText: dydxFormatter.shared.dollar(number: stopLossOrder.triggerPrice?.doubleValue, digits: decimalDigits),
+                        limitPrice: stopLossOrder.type == .stoplimit ? dydxFormatter.shared.dollar(number: stopLossOrder.price, digits: decimalDigits) : nil,
+                        // don't show amount unless order size is custom
+                        amount: positionSize == orderSize ? nil : dydxFormatter.shared.percent(number: orderSize/positionSize, digits: 2),
+                        action: routeToTakeProfitStopLossAction)
+                } else {
+                    viewModel?.stopLossStatusViewModel = .init(
+                        triggerSide: .stopLoss,
+                        action: routeToTakeProfitStopLossAction)
+                }
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift]: new logic for display different states for this area
Screenshot 2024-04-29 at 4 48 06 PM

File: dydx/dydxStateManager/dydxStateManager/AbacusState+Combine.swift [hunk 3]
```diff
@@ -200,6 +200,18 @@ public final class AbacusState {
             .eraseToAnyPublisher()
     }
 
+    public var selectedSubaccountTriggerOrders: AnyPublisher<[SubaccountOrder], Never> {
+        selectedSubaccountOrders
+            .map { orders in
+                orders.filter { order in
+                    order.status == .untriggered
```
Comment (mike-dydx) [inline dydx/dydxStateManager/dydxStateManager/AbacusState+Combine.swift:207]: thanks for this suggestion @ruixhuang

File: dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift (hunk 1 above)
Comment (ruixhuang) [reply dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift:97]: Yeah.. I need to add it to Android as well

File: dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift [hunk 4]
```diff
@@ -78,69 +78,78 @@ class dydxMarketPositionViewPresenter: HostedViewPresenter<dydxMarketPositionVie
...
+                        triggerSide: .takeProfit,
+                        triggerPriceText: DataLocalizer.shared?.localize(path: "APP.TRADE.MULTIPLE_ARROW", params: nil),
+                        action: routeToOrdersAction)
+                } else if let takeProfitOrder = takeProfitOrders.first, let positionSize = position.size?.current?.doubleValue.magnitude {
+                    let orderSize = takeProfitOrder.size.magnitude
+                    viewModel?.takeProfitStatusViewModel = .init(
+                        triggerSide: .takeProfit,
+                        triggerPriceText: dydxFormatter.shared.dollar(number: takeProfitOrder.triggerPrice?.doubleValue, digits: decimalDigits),
+                        limitPrice: takeProfitOrder.type == .takeprofitlimit ? dydxFormatter.shared.dollar(number: takeProfitOrder.price, digits: decimalDigits) : nil,
+                        amount: positionSize == orderSize ? nil : dydxFormatter.shared.percent(number: orderSize/positionSize, digits: 2),
```
Comment (ruixhuang) [inline dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift]: check for positionSize != 0?

Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift]: yeah will add to be safe 👍

Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift]: addressed

File: dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewModel.swift [hunk 5]
```diff
@@ -115,7 +116,7 @@ public class dydxTakeProfitStopLossViewModel: PlatformViewModel {
                 .themeFont(fontType: .base, fontSize: .small)
             Rectangle()
                 .frame(height: 1)
-                .themeColor(background: .layer6)
+                .overlay(ThemeColor.SemanticColor.layer6.color)
```
Comment (ruixhuang) [inline dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewModel.swift:119]: Does this make a difference?

Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewModel.swift:119]: yeah, setting background color on a rectangle doesn't do anything
Screenshot 2024-04-29 at 7 15 37 PM

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**